app/models/model.rb

```diff
@@ -18,4 +18,8 @@ class Model < ApplicationRecord
   scoped_search :on => :info
   scoped_search :on => :vendor_class, :complete_value => :true
   scoped_search :on => :hardware_model, :complete_value => :true
+
+  def hosts_count
+    Host::Managed.select(:id).where(:model_id => id).count
```

Comment:
maybe this can be includede in the has_many_hosts? (https://github.com/theforeman/foreman/blob/develop/app/models/host/hostmix.rb)